### PR TITLE
Add go.mod

### PR DESCRIPTION
**Pull request by ashokponkumar**
Signed-off-by: Ashok Pon Kumar <ashokponkumar@gmail.com>

### CODE REVIEW

**Comment by jdolitsky** [comment]
@ashokponkumar was this required for you to use some of the Go libs in this repo?

**Comment by ashokponkumar** [comment]
> @ashokponkumar was this required for you to use some of the Go libs in this repo?

Not for this repo specifically. I am using the runtime-spec in https://github.ibm.com/app-modernization/move2kube/blob/master/go.mod and I noticed that some of the other spec's were missing the go.mod and that it might create a problem once we hit v2, hence added them and raised the pull request.

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,3 @@
+module github.com/opencontainers/distribution-spec
```
Comment (mikebrow) [inline go.mod:1]: Should this be based off `go mod init specs-go/specs` As there are no go files in the root or packages..

resulting in `module specs-go/specs`

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,3 @@
...
+
+go 1.14
```
Comment (mikebrow) [inline go.mod:3]: should either move up travis to 1.14 or 15 or change this to go 1.13 to match travis test.. See: https://github.com/opencontainers/distribution-spec/blob/master/.travis.yml#L3

**Review by mikebrow — COMMENTED** [review]
see questions comments...

File: go.mod (hunk 1 above)
Comment (mikebrow) [reply go.mod:1]: have you tested it over in another repo? I didn't try this out..

**Comment by ashokponkumar** [comment]
Thanks @mikebrow for the comments. @thaJeztah has created a proposal for a similar PR in runtime-spec in https://github.com/opencontainers/runtime-spec/issues/1069. Based on how that proposal evolves, I can appropriately make the changes here.

File: go.mod (hunk 2 above)
Comment (ashokponkumar) [reply go.mod]: Sure @mikebrow. Have moved both to 1.15

File: go.mod (hunk 1 above)
Comment (ashokponkumar) [reply go.mod:1]: We can evolve it similar to https://github.com/opencontainers/runtime-spec/issues/1069

**Comment by vbatts** [comment]
Is this still needed?

**Comment by thaJeztah** [comment]
I think this is still desirable, but some discussion needed on the approach (`go.mod` in subdirectory, not at the top, and potentially separate versioning to allow fixes that only affect the go-code, not the spec itself); https://github.com/opencontainers/distribution-spec/pull/192#issuecomment-702513616
 
> Thanks @mikebrow for the comments. @thaJeztah has created a proposal for a similar PR in runtime-spec in [opencontainers/runtime-spec#1069](https://github.com/opencontainers/runtime-spec/issues/1069). Based on how that proposal evolves, I can appropriately make the changes here.

**Review by vbatts — APPROVED** [review]
🎇

**Comment by vbatts** [comment]
maybe better if someone carries this and put's the go.mod in the right place

**Comment by vbatts** [comment]
closing in favor of #283

**Pull request closed without merging**